network/httpd_proxy.py

```diff
@@ -0,0 +1,344 @@
+        (Requires member_set_status: true).
+    required: false
+    default: false
+  ignore_errors:
+    description:
+      - Ignore all HTTP replies errors from the member, preventing automatic disabling of this member from the balancer pool.
+        (Requires member_set_status: true).
+    required: false
+    default: false
+  member_host:
+    description:
+      - [ipv4|ipv6|fqdn] of the balancer member to get or to set attributes to.
+        Port number is autodetected and should not be specified here.
+        If undefined, httpd_proxy module will return a "members" list of dictionaries of all the current balancer pool members' attributes.
+    required: false
+    default: None
+  member_set_status:
+    description:
+      - Flag needed to set a balancer pool member status.
+        (Requires member_host).
+    required: false
+    default: false
+'''
+
+EXAMPLES = '''
+# Get all current balancer pool members' attributes:
+- httpd_proxy: balancer_vhost=10.0.0.2
+
+# Get a specific member's attributes:
+- httpd_proxy: balancer_vhost=myws.mydomain.org balancer_suffix="/lb/" member_host=node1.myws.mydomain.org
+
+# Enable all balancer pool members:
+- httpd_proxy: balancer_vhost="{{ myloadbalancer_ip }}"
+  register: result
+- httpd_proxy: balancer_vhost="{{ myloadbalancer_ip }}" member_host="{{ item.ip }}" member_set_status=true enabled=true
+  with_items: "{{ result.members }}"
+
+# Gracefully disable a member from a loadbalancer node:
+- httpd_proxy: balancer_vhost="{{ vhost_ip }}" member_host="{{ member.ip }}" member_set_status=true draining_mode=true delegate_to=myloadbalancernode
+- wait_for: host="{{ member.ip }}" port={{ member.port}} state=drained delegate_to=myloadbalancernode
+- httpd_proxy: balancer_vhost="{{ vhost_ip }}" member_host="{{ member.ip }}" member_set_status=true disabled=true delegate_to=myloadbalancernode
+
+# "member" dictionary output example, when specifying a member node:
+{
+    "attributes": {
+        "Busy": "0",
+        "Elected": "42",
+        "Factor": "1",
+        "From": "136K",
+        "Load": "0",
+        "Route": null,
+        "RouteRedir": null,
+        "Set": "0",
+        "Status": "Init Ok ",
+        "To": " 47K",
+        "Worker URL": null
+    },
+    "balancer_url": "http://10.10.0.2/balancer-manager/",
+    "ip": "10.10.0.20",
+    "management_url": "http://10.10.0.2/lb/?b=mywsbalancer&w=http://10.10.0.20:8080/ws&nonce=8925436c-79c6-4841-8936-e7d13b79239b",
+    "path": "/ws",
+    "port": 8080,
+    "status": {
+        "disabled": false,
+        "draining_mode": false,
+        "hot_standby": false,
+        "ignore_errors": false
+    }
+}
+
+author: "Olivier Boukili (boukili.olivier@gmail.com)"
+'''
+
+import re
+
+try:
+    from BeautifulSoup import BeautifulSoup
+except ImportError as exception:
+    HAS_BEAUTIFULSOUP = False
+else:
+    HAS_BEAUTIFULSOUP = True
+
+try:
+    import requests
```

Comment:
Would it be possible to let this use ansibles url handling instead of requests? That would avoid a req on the managed nodes.